### PR TITLE
45996216 Classifying the Alzheimer’s disease based on the Siamese network

### PR DESCRIPTION
**Pull request by Xinyan44**
Student Number: 45996216
Name: Jialiang Hou

this pull request is consist of:

dataset.py: The method of preprocessing and loading train and test data
module.py: The construction of the neural network
train.py: Training and saving model
predict.py: Report the train and test accuracy
demo.ipynb: A demo to show the program
README.md: Illustrating the problem and the solution

The README.md gives instructions on how to reproduce the results. All the necessary files except the dataset are uploaded.
Thanks,
Gallen

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
- Commit log is very short and messages are not descriptive
- Readme is not satisfactory. No input, output, results plots shown (images don't seem to be showing up due to absolute paths used). 
- Looking at the ipynb file, classification accuracy seems great (99% train and 97% for test) - although such high numbers are suspicious for a difficult dataset. Backbone seems very simple too.
- After taking another look at the code, the classification performed seems to be between pairs of images for their similarity for each iteration, hence the high accuracy. No contrastive loss implemented for the siamese, instead a classification loss is used for the similarity of images. This is incorrect according to the requirements of the solution. The classifier needs to take in embeddings of individual images from the Siamese and classify their classes separately, not their similarity in pairs.
- Some example prediction results (for pairs of images) given in the ipynb file. From the implementation, prediction can only be performed for pairs of images.

**Comment by Xinyan44** [comment]
Hello, Nathasha,
Thanks for the informative feedback. From your feedback, I've known there are some drawbacks within my project and I can improve my report from these perspectives. First I wanna apologise for the short commit log, it mainly because I implement the code locally and upload it at one time. 
There are some issues in the input and output results polts, I tested it on my local machine, and it seems to work correctly but there is truly some error to show the plots. But all the results are uploaded as well. You could look at the images in the folder.
Thanks again for answering some questions in my mind and giving me a better understanding of how should we predict images by using the Siamese network. I am not true if it is correct or not but I think this network can predict a single image. It just needs to put two images into the model where one image we already know the label. The model will return whether they are in the same class. So we can get the label of the image we need to predict. From my point of view, this is equivalent to predicting a single image. Although this model can only predict pairs of images, one image can randomly choose from the training set.

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation (some issues found) -2
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem (seems there are issues in the solution, see comments above) -4
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage (no plots, outputs etc.) -2
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages (too short in general) -2
Progressive commits used (too few commits, commit more regularly next time) -2
## Documentation
ReadMe little detail -2
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove pyc cache files as they take up repo space. -2
Request Description good

**Comment by Xinyan44** [comment]
## Name: Jialiang Hou; Student Number: 45996216

> ## Change log
> Remove the cache pyc files from the folder
> add more detail to the Readme File
> add more comments to the code
